### PR TITLE
Installed eslint, added eslint rules and lint script with --fix flag

### LINKED ISSUE #5: Install eslint and copy rules from ai-battle-website
We need consistent code!
- Install [eslint](https://www.npmjs.com/package/eslint) and save it as a package dependency.
- Add the rules from the [ai-battle-website](https://github.com/JSJitsu/ai-battle-website/) repository to this one.
- Ensure the linter can be run by adding a "lint" script to _package.json_.

Bonus: Use eslint to automatically fix existing linting issues.

### PR DESCRIPTION
**Pull request by dmnkgrc**
Should fix #5 :)

### CODE REVIEW

**Comment by simeg** [comment]
Woah, 3425 files changed. You should split up your changes into multiple commits so it's easier to review this PR :)

**Comment by Asuza** [comment]
@dmnkgrc, thanks for the PR! I probably should have been more specific in mentioning to not check in the _node_modules_ directory. I already have _.gitignore_ files in other repos to ignore it and didn't think about it.

Would you please update the PR to **not** include the _node_modules_ folder? In case I didn't mention it earlier, eslint should be able to be installed by running `npm install`.

**Comment by dmnkgrc** [comment]
Sorry, didn't realize there wasn't a .gitignore

File: npm-debug.log [hunk 1]
```diff
@@ -0,0 +1,48 @@
+0 info it worked if it ends with ok
```
Comment (Asuza) [inline npm-debug.log]: Please remove this file and add it to _.gitignore_. Don't want to store logs in the repo. :smile:

File: package.json [hunk 2]
```diff
@@ -4,7 +4,8 @@
   "description": "Engine used by various sites to power the games",
   "main": "index.js",
   "scripts": {
-    "test": "npm test"
+    "test": "npm test",
+    "lint": "eslint index.js lib/game_classes/ --fix"
```
Comment (Asuza) [inline package.json]: Please change the `lint` script to work on all the files, and not fix them automatically. I believe it would be:

```
eslint ./
```

**Comment by Asuza** [comment]
@dmnkgrc, this PR is looking really good. If you can make the two adjustments I mentioned above, and fix the merge conflict, I believe it will be ready to merge. Thanks!

**Comment by dmnkgrc** [comment]
Done :)

**Comment by Asuza** [comment]
Thanks, @dmnkgrc!

**Pull request merged**